README.md

```diff
@@ -30,18 +29,76 @@ All scripts should be executed in the virtual environment managed by `poetry`.
 
-- `show_simulated_subgraph.py` - runs and (optionally) visualizes the simulation (only environment)
-- `show_bandit.py` - runs and (optionally) visualizes the simulation (both agent and environment)
-- `train_bandit.py` - runs the simulation and trains an agent in a given environment (no visualization, logging to TensorBoard)
+**Note:** We do not use default values _anywhere_ in our code.
+Every value must come from the config file.
 
-Each script can be parametrized by arguments passed from the command line. For example, to run the simulation with `ppo` agent in the `noisy_static` environment please run:
+#### Simulation Config
 
-```console
-poetry run python bandit_scripts/show_bandit.py -a ppo -e noisy_static --show
+The simulation config must be a python file.
+Let's break down the various components of the simulation config.
+
+The first thing you need to do is define a function that is captured by the simulation ingredient.
+The name of the function itself doesn't matter, but we use `config` here for clarity.
+
+``` python
+from simulation import simulation_ingredient  # Import the simulation ingredient
+
+@simulation_ingredient.config  # Tag as the simulation config
+def config():
+    ...
+```
+
+We use the simulation config to construct the simulation `Environment`.
+As such, the config should specify the inputs to the `Environment` class' initialiser.
+
+``` python
+"""
+    distributor (dict[str, Any]): The config for the query distributor.
+    entities (list[dict[str, Any]]): The configs for each group of entities.
+    nepisodes (int): How many episodes to run.
+    ntimesteps (int): How many timesteps to run each episode for.
+"""
+```
+
+`nepisodes` and `ntimesteps` are self-explanatory.
+
+`distributor` is a dictionary that specifies the configuration of which ISA to use.
+See the `distributor` documentation for more details.
+
+`entities` is a list of configs for each entity type.
+Each entry in the `entities` list is a dictionary.
+The dictionary can be of kind `entity` in which case the entity has only a state, but no action, or `agent` in which case the entity has a state and an action.
+Check out the `entity` documentation for more details.
+
+#### Algorithm Config
+
+Similarly to the simulation config, the first thing you need to do is define a function that is captured by the algorithm ingredient.
+
+``` python
+from autoagora_agents import algorithm_ingredient  # Import the algorithm ingredient
+
+@algorithm_ingredient.config  # Tag as the algorithm config
+def config():
+    ...
 ```
 
-More details on arguments can be accessed in help (`--h`).
+The experiment uses the algorithm config to construct the `Controller` object, which maps the simulation to the algorithms.
+The `Controller` also constructs the agents.
+It takes two inputs: `seed` and `agents`.
+
+The `seed` is just the random seed set for reproducibility.
```

Comment:
shouldn't that be part of `experiment` section/config?

Reply:
Currently, only the algorithm uses the `seed`. If there were an `experiment` config, I'd agree with you. Since there isn't either we put it in two places - `simulation` and `algorithm`, or we just leave it the way it is. Unless you want me to create an experiment config that just holds the seed?

Reply:
Went backwards through these haha. I now see where the `experiment` config came in.

Reply:
https://github.com/semiotic-ai/autoagora-agents/pull/42/commits/3aab0221ad5bc1073757b632248063fa612778d2